drivers/gnss/gnss_seeed_air530z.c

```diff
@@ -0,0 +1,332 @@
+	data->dynamic_script_chat.request = data->dynamic_request_buf;
+	data->dynamic_script_chat.response_matches = NULL;
+	data->dynamic_script_chat.response_matches_size = 0;
+	data->dynamic_script_chat.timeout = 0;
+
+	data->dynamic_script.name = "PCAS";
+	data->dynamic_script.script_chats = &data->dynamic_script_chat;
+	data->dynamic_script.script_chats_size = 1;
+	data->dynamic_script.abort_matches = NULL;
+	data->dynamic_script.abort_matches_size = 0;
+	data->dynamic_script.callback = NULL;
+	data->dynamic_script.timeout = 5;
+}
+
+static int gnss_seeed_air530z_init(const struct device *dev)
+{
+	struct gnss_seeed_air530z_data *data = dev->data;
+	const struct gnss_seeed_air530z_config *config = dev->config;
+	int ret;
+
+	LOG_DBG("initializing SEED Air530z");
+
+	ret = gnss_seeed_air530z_init_nmea0183_match(dev);
+	if (ret < 0) {
+		LOG_ERR("ERROR gnss_seeed_air530z_init_nmea0183_match");
+		return ret;
+	}
+
+	gnss_seeed_air530z_init_pipe(dev);
+
+	ret = gnss_seeed_air530z_init_chat(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	seeed_air530z_init_dynamic_script(dev);
+
+	ret = modem_pipe_open(data->uart_pipe);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = modem_chat_attach(&data->chat, data->uart_pipe);
+	if (ret < 0) {
+		modem_pipe_close(data->uart_pipe);
+		return ret;
+	}
+
+	ret = modem_chat_run_script(&data->chat, &init_script);
+	if (ret < 0) {
+		LOG_ERR("Failed to run init_script");
+		modem_pipe_close(data->uart_pipe);
+		return ret;
+	}
+
+	/* setup on-off gpio for power management */
+	if (!gpio_is_ready_dt(&config->on_off_gpio)) {
+		LOG_ERR("on-off GPIO device not ready");
+		return -ENODEV;
+	}
+
+	gpio_pin_configure_dt(&config->on_off_gpio, GPIO_OUTPUT_LOW);
+
+	return 0;
+}
+
+static int seeed_air530z_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	struct gnss_seeed_air530z_data *data = dev->data;
+	const struct gnss_seeed_air530z_config *config = dev->config;
+	k_spinlock_key_t key;
+	int ret = -ENOTSUP;
+
+	key = k_spin_lock(&data->lock);
+
+	switch (action) {
+	case PM_DEVICE_ACTION_SUSPEND:
+		LOG_INF("Suspending");
+		gpio_pin_set_dt(&config->on_off_gpio, 0);
+		ret = 0;
+		break;
+
+	case PM_DEVICE_ACTION_RESUME:
+		LOG_INF("Resumming");
+		gpio_pin_set_dt(&config->on_off_gpio, 1);
+		ret = 0;
+		break;
+
+	default:
+		break;
+	}
+
+	k_spin_unlock(&data->lock, key);
+	return ret;
+}
+
+static int seeed_air530z_set_fix_rate(const struct device *dev, uint32_t fix_interval_ms)
+{
+	struct gnss_seeed_air530z_data *data = dev->data;
+	k_spinlock_key_t key;
+	int ret;
+
+	if (fix_interval_ms < 100 || fix_interval_ms > 1000) {
+		return -EINVAL;
+	}
+
+	key = k_spin_lock(&data->lock);
+
+	ret = gnss_nmea0183_snprintk(data->dynamic_request_buf, sizeof(data->dynamic_request_buf),
+				    "PCAS02,%u", fix_interval_ms);
+
+	data->dynamic_script_chat.request_size = ret;
+
+	ret = modem_chat_run_script(&data->chat, &data->dynamic_script);
+
+	if (ret < 0) {
+		goto unlock_return;
+	}
```

Comment:
```suggestion
	ret = modem_chat_run_script(&data->chat, &data->dynamic_script);
	if (ret < 0) {
		goto unlock_return;
	}
```

it's common practice to keep calls and their check together, same below

Reply:
Got it!